src/components/forms/select/RuiMenuSelect.vue

```diff
@@ -29,7 +28,9 @@ export interface Props {
   hint?: string;
   errorMessages?: string | string[];
   successMessages?: string | string[];
-  showDetails?: boolean;
+  returnPrimitive?: boolean;
```

Comment:
Can we avoid this as a property? It should be better handled internally.

Basically what we can do is make the key optional, and if the key exists return data[key] otherwise return the whole item. I did remove the same property from the AutoComplete component we have in the main library.